internal/k6runner/k6runner.go

```diff
@@ -328,29 +344,98 @@ type RunResponse struct {
 }
 
 func (r HttpRunner) WithLogger(logger *zerolog.Logger) Runner {
-	return HttpRunner{
-		url:    r.url,
-		logger: logger,
-	}
+	r.logger = logger
+
+	return r
 }
 
 var ErrUnexpectedStatus = errors.New("unexpected status code")
 
 func (r HttpRunner) Run(ctx context.Context, script Script) (*RunResponse, error) {
-	k6Timeout := time.Duration(script.Settings.Timeout) * time.Millisecond
+	if r.backoff == 0 {
+		panic("zero backoff, runner is misconfigured, refusing to DoS")
+	}
+
+	if deadline, hasDeadline := ctx.Deadline(); !hasDeadline {
+		defaultAllRetriesTimeout := time.Duration(script.Settings.Timeout) * time.Millisecond * 2
+		r.logger.Error().
```

Comment:
This is the type of fallback that is there just to prevent a deadlock, but should not be relied upon. If we see this in production, it means there's a bug in our code and needs attention.